### PR TITLE
Remove apps & incentivized channel from parachain

### PR DESCRIPTION
**Pull request by doubledup**
Builds pass for all runtimes, the query-events tool and the test parachain. Not sure I've caught everything that can be removed in this pass. Notably the assets and treasury pallets can probably be removed with the move to Statemint and BridgeHub.

This is merging into a feature branch to keep the PRs (relatively) small. I'll make a PR for the E2E tests once the relayer & contracts are updated.

### CODE REVIEW

File: parachain/primitives/core/Cargo.toml [hunk 1]
```diff
@@ -8,7 +8,6 @@ edition = "2021"
 serde = { version = "1.0.137", optional = true, features = [ "derive" ] }
 codec = { package = "parity-scale-codec", version = "3.1.5", default-features = false }
 scale-info = { version = "2.2.0", default-features = false, features = [ "derive" ] }
-enum-iterator = "0.6.0"
```
Comment (claravanstaden) [inline parachain/primitives/core/Cargo.toml:11]: Is this an unrelated, unused module?

File: parachain/runtime/snowbase/src/lib.rs [hunk 2]
```diff
@@ -788,8 +709,7 @@ construct_runtime!(
 		// Bridge Infrastructure
 		BasicInboundChannel: basic_channel_inbound::{Pallet, Call, Config, Storage, Event<T>} = 12,
 		BasicOutboundChannel: basic_channel_outbound::{Pallet, Config<T>, Storage, Event<T>} = 13,
-		IncentivizedInboundChannel: incentivized_channel_inbound::{Pallet, Call, Config, Storage, Event<T>} = 14,
-		IncentivizedOutboundChannel: incentivized_channel_outbound::{Pallet, Call, Config<T>, Storage, Event<T>} = 15,
+		// 14 and 15 were used for the incentivized inbound and outbound channels respectively
```
Comment (claravanstaden) [inline parachain/runtime/snowbase/src/lib.rs:712]: Useful comment! 👍

File: parachain/primitives/core/Cargo.toml (hunk 1 above)
Comment (doubledup) [reply parachain/primitives/core/Cargo.toml:11]: yup, we were only [using it for `ChannelId`](https://github.com/Snowfork/snowbridge/pull/753/files#diff-0a114baf0463c3e1f1e8522bf78e02c55ce5fb44fef902204ba4caf5f6e3b951L4), so it's a knock-on effect of removing `ChannelId`.

**Review by alistair-singh — APPROVED** [review]
Looks good!

**Review by vgeddes — APPROVED** [review]
+1

**Pull request merged**